src/Microsoft.AspNetCore.JsonPatch/Helpers/ExpressionHelpers.cs

```diff
@@ -93,6 +70,26 @@ private static string GetPath(Expression expr, bool firstTime)
             }
         }
 
+        private static string GetPropertyNameFromMemberExpression(MemberExpression memberExpression)
+        {
+            // if there's a JsonProperty attribute, we must return the PropertyName
+            // from the attribute rather than the member name 
+            var jsonPropertyAttribute =
+                memberExpression.Member.GetCustomAttributes(
+                typeof(JsonPropertyAttribute), true);
+
+            // to object[] to avoid LINQ
+            var jsonPropertyAttributeAsObjectArray = (object[])jsonPropertyAttribute;
```

Comment:
This is not OK :laughing: - we shouldn't do a downcast here because this is a framework implementation detail.

You should be able to use https://msdn.microsoft.com/en-us/library/hh138307(v=vs.110).aspx here and avoid linq and downcasting